tests/ops/functions/test_matrix.py

```diff
@@ -0,0 +1,580 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for the get_unitary_matrix transform
+"""
+from functools import reduce
+
+import pytest
+
+import pennylane as qml
+from pennylane import numpy as np
+from pennylane.transforms.op_transforms import OperationTransformError
+
+from gate_data import I, X, Y, Z, H, S, CNOT, Roty as RY
+
+one_qubit_no_parameter = [
+    qml.PauliX,
+    qml.PauliY,
+    qml.PauliZ,
+    qml.Hadamard,
+    qml.S,
+    qml.T,
+    qml.SX,
+]
+
+
+one_qubit_one_parameter = [qml.RX, qml.RY, qml.RZ, qml.PhaseShift]
+
+
+class TestSingleOperation:
+    @pytest.mark.parametrize("op_class", one_qubit_no_parameter)
+    def test_non_parametric_instantiated(self, op_class):
+        """Verify that the matrices of non-parametric one qubit gates is correct
+        when provided as an instantiated operation"""
+        op = op_class(wires=0)
+        res = qml.matrix(op)
+        expected = op.get_matrix()
+        assert np.allclose(res, expected)
+
+    @pytest.mark.parametrize("op_class", one_qubit_no_parameter)
+    def test_non_parametric_gates_qfunc(self, op_class):
+        """Verify that the matrices of non-parametric one qubit gates is correct
+        when provided as a qfunc"""

```

Comment:
I may be blind, but it's more provided as a class, not a qfunc, no?

Reply:
instantiating an operator is basically the same thing, from PL's point of view, as a qfunc that only queues a single operation 😆 Which is very useful way of thinking when writing the tests